proposals/accepted/000-ast-parser-libs.rst

```diff
@@ -0,0 +1,344 @@
+
+The Trees That Grow project is now 6 years old, and has met great success in avoiding partiality in the compiler, "making illegal states unrepresentable" as many Haskellers would put it.
+But progress on `reducing AST & parser dependencies <https://gitlab.haskell.org/ghc/ghc/-/issues/19932>`_ has been less easily forthcoming.
+We have separated out the modules defining the AST under ``Language.Haskell.Syntax.*`` we wish to split out, and we have tests to track progress reducing their deps, and the parser's deps.
+But progress is unsteady and unpredictable.
+
+The basic problem is that the benefits don't actually kick in until the deps are *all* gone, and the code is actually separated out.
+Partial progress isn't really directly useful to anyone, and these counters just scoreboard by which we hope to get closer to the end goal.
+It is thus hard to do this work with volunteers only, because it is emphatically *not* `"itch scratching" <https://en.wikipedia.org/wiki/The_Cathedral_and_the_Bazaar>`_ work where incremental progress leads immediate incremental benefits to the contributor.
+
+The Haskell Foundation's support in getting this "over the finish line", at which point the community *will* benefit, and benefit greatly, is thus a crucial way we can surmount the coordination failure the lack of incremental payoff causes.
+
+.. [#intra]
+  It might sound like the goal is only different usages within GHC, but remember that ``template-haskell`` is a separate library used by users of Haskell not just developers of Haskell.
+  A goal of at least some usage outside GHC was always there.
+
+Reinstallable GHC Lib
+---------------------
+
+One of the problems ``ghc-lib-parser`` aims to solve is that ``ghc`` the library is current cumbersome to install as a regular haskell library (as opposed to by switching toolchains entirely).
+There is currently work in flight to solve that.
+One that is done, projects like HLint_ *could* just depend on ``ghc`` directly, and still be easily buildable (with Cabal / with Stack / from Hackage) as today.
+
+Just doing this isn't a good solution though, because ``ghc`` exposes a much a wider surface area than what these projects actually want.
+For stability's sake, it is better that those libraries dependent on narrower parsing / AST libraries that only provide what they need.
+And longer term, we hope the "tug of war" of between GHC and these projects as consumers of those libraries, versus just the others having to deal with whatever GHC does with just itself in mind, will result in a higher-quality, more flexible, and overall friendlier library.
+
+In `this comment <https://gitlab.haskell.org/ghc/ghc/-/issues/14409#note_506489>`_, it is suggested that factoring out the AST and parser can be a good first step making a more modular in GHC in general.
+This proposal wish to *stay neutral* on the merits of such a future direction, but it would be remiss not to at least highlight it as one possible outcome.
```

Comment:
```suggestion
This proposal wishes to *stay neutral* on the merits of such a future direction, but it would be remiss not to at least highlight it as one possible outcome.
```